runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -599,7 +606,7 @@ private Object applyUpdateUnsafe(SMREntry entry, long timestamp) {
             // If there was no previously calculated undo entry
             // The undo records shouldn't be computed for objects with a MONOTONIC version
             // policy because they simply don't sync backwards to older versions.
-            if (undoRecordTarget != null && object.getVersionPolicy() != ICorfuVersionPolicy.MONOTONIC) {
+            if (undoRecordTarget != null && !isMonotonicObject()) {
```

Comment:
It looks like this check can be removed now since it's captured in the calling method.

Reply:
Ignore this comment. It is still required.